### PR TITLE
Update setup.py to allow for dbt-adapters ~1.3

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
Resolves #

### Checklist

- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

**Comment by benc-db** [comment]
Running functional tests now

File: setup.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def _get_plugin_version() -> str:
     install_requires=[
         "dbt-spark>=1.8.0, <2.0",
         "dbt-core~=1.8.0",
-        "dbt-adapters~=1.2.0",
+        "dbt-adapters~=1.3.0",
```
Comment (ChenyuLInx) [inline setup.py:59]:
```suggestion
        "dbt-adapters~=1.2",
```

Can we do just this so we don't need to bump next time we do a minor release?

Comment (benc-db) [reply setup.py]: Ok, but if you violate SemVer...

File: setup.py [hunk 2]
```diff
@@ -56,7 +56,7 @@ def _get_plugin_version() -> str:
     install_requires=[
         "dbt-spark>=1.8.0, <2.0",
         "dbt-core~=1.8.0",
-        "dbt-adapters~=1.2.0",
+        "dbt-adapters>=1.2.0, <2.0",
```
Comment (aranke) [inline setup.py:59]:
```suggestion
        "dbt-adapters>=1.3.0, <2.0",
```

for `get_catalog_for_single_relation` from 1.3.0 ([link](https://github.com/dbt-labs/dbt-adapters/blob/b64d3a5efd83b8a3ab0612145f6bdeaa8cb0f2ba/.changes/1.3.0.md#L3))

File: setup.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def _get_plugin_version() -> str:
     install_requires=[
         "dbt-spark>=1.8.0, <2.0",
         "dbt-core~=1.8.0",
```
Comment (aranke) [inline setup.py:58]: For consistency, could you also bump this pin?

```suggestion
        "dbt-core>=1.8.0, <2.0",
```

**Comment by benc-db** [comment]
@colin-rogers-dbt please also update requirements.txt

**Comment by benc-db** [comment]
Will run functional tests now

File: requirements.txt [hunk 4]
```diff
@@ -1,8 +1,8 @@
 databricks-sql-connector>=3.1.0, <3.2.0
 dbt-spark~=1.8.0
-dbt-core~=1.8.0
-dbt-adapters~=1.2.0
+dbt-core>=1.8.0, <2.0
+dbt-adapters>=1.3.0, <2.0
 databricks-sdk==0.17.0
 keyring>=23.13.0
 pandas<2.2.0
```
Comment (aranke) [inline requirements.txt:7]: This whole file could just be `.`

Comment (benc-db) [reply requirements.txt:7]: Will investigate that as a separate PR, thanks for bringing to my attention

**Pull request merged**